### DISCUSSION TITLE
Can we see which non-FQ rule matches in a filter?

### DISCUSSION THREAD

**Discussion by TonyGravagno** [RealRaven2000/FiltaQuilla#279] [Q&A]
FQ has nice console.logging to show a rule, the search term (value being queried), and the result:

> FiltaQuilla
>   subjectRegex RESULT: true 
>   search term: Subject matches domain\\.tld filtaquilla-util.js:255:12

Can we get standard Tb filters to report the success/failure of its own rules?

Might there be events that broadcast this information to listeners that might be added to FQ?

TY

**Comment by RealRaven2000** [comment]
> Can we get standard Tb filters to report the success/failure of its own rules?

I am afraid this isn't possible. As far as my knowledge goes there are simply no entry points for Add-ons to modify existing filter actions or search terms, as there is no interface in the scriptable parts of the filter service to do so. We can only create custom actions and search terms which implementation we can fully control.

If you would like to do some digging here are the current interfaces for the filter service - this can only be accessed via legacy code that has access to the thunderbird core JavaScript layer (it's complex):
[nsIMsgFilterService](https://searchfox.org/comm-esr128/source/mailnews/search/public/nsIMsgFilterService.idl)
[nsIMsgFilter](https://searchfox.org/comm-esr128/source/mailnews/search/public/nsIMsgFilter.idl)

there is currently no filters object in the web extensions API (it is very hard to come up with interfaces that break it down to a useful level on Add-on perspective)
